drivers/flash/spi_flash_at45.c

```diff
@@ -0,0 +1,685 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/flash.h>
+#include <drivers/spi.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(spi_flash_at45, CONFIG_FLASH_LOG_LEVEL);
+
+/* AT45 commands used by this driver: */
+/* - Continuous Array Read (Low Power Mode) */
+#define CMD_READ		0x01
+/* - Main Memory Byte/Page Program through Buffer 1 without Built-In Erase */
+#define CMD_WRITE		0x02
+/* - Read-Modify-Write */
+#define CMD_MODIFY		0x58
+/* - Manufacturer and Device ID Read */
+#define CMD_READ_ID		0x9F
+/* - Status Register Read */
+#define CMD_READ_STATUS		0xD7
+/* - Chip Erase */
+#define CMD_CHIP_ERASE		{ 0xC7, 0x94, 0x80, 0x9A }
+/* - Sector Erase */
+#define CMD_SECTOR_ERASE	0x7C
+/* - Block Erase */
+#define CMD_BLOCK_ERASE		0x50
+/* - Page Erase */
+#define CMD_PAGE_ERASE		0x81
+/* - Deep Power-Down */
+#define CMD_ENTER_DPD		0xB9
+/* - Resume from Deep Power-Down */
+#define CMD_EXIT_DPD		0xAB
+/* - Ultra-Deep Power-Down */
+#define CMD_ENTER_UDPD		0x79
+/* - Buffer and Page Size Configuration, "Power of 2" binary page size */
+#define CMD_BINARY_PAGE_SIZE	{ 0x3D, 0x2A, 0x80, 0xA6 }
+
+#define AT45_SECTOR_SIZE	0x10000UL
+
+#define STATUS_REG_LSB_RDY_BUSY_BIT	0x80
+#define STATUS_REG_LSB_PAGE_SIZE_BIT	0x01
+
+
+#define DEF_BUF_SET(_name, _buf_array) \
+	const struct spi_buf_set _name = { \
+		.buffers = _buf_array, \
+		.count   = ARRAY_SIZE(_buf_array), \
+	}
+
+struct spi_flash_at45_data {
+	struct device *spi;
+	struct spi_cs_control spi_cs;
+	struct k_sem lock;
+#if IS_ENABLED(CONFIG_DEVICE_POWER_MANAGEMENT)
+	u32_t pm_state;
+#endif
+};
+
+struct spi_flash_at45_config {
+	const char *spi_bus;
+	struct spi_config spi_cfg;
+	const char *cs_gpio;
+	int cs_pin;
+#if IS_ENABLED(CONFIG_FLASH_PAGE_LAYOUT)
+	struct flash_pages_layout pages_layout;
+#endif
+	u32_t chip_size;
+	u16_t block_size;
+	u16_t page_size;
+	u16_t t_enter_dpd; /* in microseconds */
```

Comment:
update comment with corrected units.

Reply:
But these time values are in microseconds. They are converted from nanoseconds at [initialization](https://github.com/zephyrproject-rtos/zephyr/blob/869dfbb3f49d890546019112d881f8062de2e833/drivers/flash/spi_flash_at45.c#L665) of this structure.

Reply:
OK, fine then, thanks for the clarification.